internal/pkg/events/events.go

```diff
@@ -252,15 +259,29 @@ func StartEvents() {
 					log.Printf("Failed to parse received event:\n- error: %s\n- event: %s\n", err, event)
 				}
 
-				record, err := elasticClient.Create(event.GetIndexName(), EVENTSINDEXTYPE, event.GetRawData())
-				if err != nil {
-					applicationHealth.ElasticSearchState = 0
-					log.Printf("Failed to save event to Elasticsearch DB:\n- error: %s\n- event: %s\n", err, event)
-				} else {
-					applicationHealth.ElasticSearchState = 1
+				process := true
+				for _, handler := range handlerManager.Handlers[amqpServers[index].DataSource] {
+					if handler.Relevant(event) {
```

Comment:
If only we could rely on the messages containing a handler identifier... one can dream

Reply:
Well, technically we could, but we would be limited. The check configuration contains information about handlers which are supposed to be used. But I wanted to make the system usable for all types of handlers. If we will be able to attach metadata to collectd events (with ceilometer it should be possible IMO, but I'm not 100% sure), we won't need to iterate across all loaded handlers and just use handler name as map key ... but that is future, not present :).

Reply:
Yep seems like a lot of work we shouldn't worry about right now 🙂